lib/ansible/modules/cloud/misc/helm_cli.py

```diff
@@ -0,0 +1,542 @@
+    deploy_command += " --reset-values"
+
+    if wait:
+        deploy_command += " --wait"
+        if wait_timeout is not None:
+            deploy_command += " --timeout " + str(wait_timeout)
+
+    if force:
+        deploy_command += " --force"
+
+    if disable_hook:
+        deploy_command += " --no-hooks"
+
+    if release_values != {}:
+        try:
+            import tempfile
+        except ImportError:
+            module.fail_json(msg="Could not import the tempfile python module. Please install `tempfile` package.")
+
+        fd, path = tempfile.mkstemp(suffix='.yml')
+        with open(path, 'w') as yaml_file:
+            yaml.dump(release_values, yaml_file, default_flow_style=False)
+        deploy_command += " -f=" + path
+
+    deploy_command += " --namespace=" + release_namespace
+    deploy_command += " " + release_name
+    deploy_command += " " + chart_name
+
+    return deploy_command
+
+
+# Delete release chart
+def delete(command, release_name, purge, disable_hook):
+    if is_helm_2:
+        delete_command = command + " delete"
+    else:
+        delete_command = command + " uninstall"
+
+    if is_helm_2 and purge:
+        delete_command += " --purge"
+    elif not is_helm_2 and not purge:
+        delete_command += " --keep-history"
+
+    if disable_hook:
+        delete_command += " --no-hooks"
+
+    delete_command += " " + release_name
+
+    return delete_command
+
+
+def main():
+    global module, is_helm_2
+    module = AnsibleModule(
+        argument_spec=dict(
+            binary_path=dict(type='path'),
+            chart_ref=dict(type='str'),
+            chart_repo_url=dict(type='str'),
+            chart_repo_username=dict(type='str'),
+            chart_repo_password=dict(type='str', no_log=True),
+            chart_version=dict(type='str'),
+            release_name=dict(type='str', required=True, aliases=['name']),
+            release_namespace=dict(type='str', default='default', aliases=['namespace']),
+            release_state=dict(default='present', choices=['present', 'absent'], aliases=['state']),
+            release_values=dict(type='dict', default={}, aliases=['values']),
+            tiller_host=dict(type='str'),
+            tiller_namespace=dict(type='str', default='kube-system'),
+            update_repo_cache=dict(type='bool', default=False),
+
+            # Helm options
+            disable_hook=dict(type='bool', default=False),
+            force=dict(type='bool', default=False),
+            purge=dict(type='bool', default=True),
+            wait=dict(type='bool', default=False),
+            wait_timeout=dict(type='int', default=300),
+        ),
+        required_if=[
+            ('release_state', 'present', ['release_name', 'chart_ref']),
+            ('release_state', 'absent', ['release_name'])
+
+        ],
+        required_together=[
+            ['chart_repo_username', 'chart_repo_password']
+        ],
+        supports_check_mode=True,
```

Comment:
I'm not too certain about this - it seems as if it would always return "changed" in check mode?

Reply:
Not exactly !
In check mode, the module return changed only if
* the chart is not install
* the chart need to be upgraded
* the values changed